src/librbd/Utils.cc

```diff
@@ -51,6 +51,7 @@ std::string generate_image_id(librados::IoCtx &ioctx) {
   librados::Rados rados(ioctx);
 
   uint64_t bid = rados.get_instance_id();
+  srand(time(NULL));
```

Comment:
We shouldn't (as a library) be re-initializing the pseudo-random number generator. The real uniqueness is provided by the global unique ID of the client -- this just adds a little extra if the same client create multiple images programmatically.

Plus, since ```time``` is only accurate in seconds, in your example it would be highly probably to re-initialize the pseudo-random number generator to the same value.

Reply:
@nuyhah ping -- perhaps remove ```rand()``` with ```std::mt19937{std::random_device{}}{}```